examples/ML/other_ml_example.py

```diff
@@ -53,9 +51,10 @@ class EvalConfig:
 train_config: TrainingConfig = args.train_config
 eval_config: EvalConfig = args.eval_config
 
-print(hparams, train_config, eval_config, sep="\n")
-expected = """
-MyModelHyperParameters(batch_size=32, learning_rate=0.05, momentum=0.01)
-TrainingConfig(data_dir='data', log_dir='logs', checkpoint_dir='checkpoints')
-EvalConfig(eval_dir='eval_data')
-"""
+assert hparams == MyModelHyperParameters(
```

Comment:
This isn't quite right, but I guess it wasn't clear why I was doing it that way to begin with, so totally understandable.

The point here is that the examples, when run with no arguments, should give `expected`. This is what the `test_examples` script checks for.
But we still want the examples to be run with different arguments! Here this will raise an AssertionError, which we don't want!

Please revert these types of changes.

Reply:
I thought you would say this. The reason I changed these tests is because they fail with 3.9, due to fields being printed in a different order. Fixing this would insert unrelated boilerplate into the example code.

A compromise could be not to try to check the output of the examples in automated tests. I think this is fine because there are plenty of actual tests for this library.

What do you think?

Reply:
Ohh the ordering of the fields is different in 3.9? that's interesting! Do you know what caused that change, out of curiosity?
Simplest solution to me seems like just xfail-ing the test in `test_examples.py` for python 3.9.
Another idea could be to just check that the word counts match instead of checking for equality when using python 3.9, and keep checking for equality with other versions!
Something like which would make the test order-agnostic, just for python 3.9:
```python3
from collections import Counter
assert Counter(actual_string.split()) == Counter(expected_string.split())
```

Reply:
I vaguely recall something about the internal implementation of dictionaries changing, and who knows what part of the code we're using here, is built on top. But I didn't look into it in detail really.

Counting words just means relying on strings again, and I'm not really sure this provides any value. I'd rather just skip the tests for 3.9. The examples are mostly simple and are covered by proper tests anyway.